src/pages/VaccineAdminPage.js

```diff
@@ -94,29 +130,84 @@ const VaccineAdminPageComponent = ({
     <DataTablePageView>
       <View style={globalStyles.pageTopSectionContainer}>
         <ToggleBar toggles={toggles} />
-        <SearchBar onChangeText={onFilterData} value={searchTerm} placeholder={placeholderString} />
-        <PageButton text="" />
+        <SearchBar
+          viewStyle={localStyles.searchBar}
+          onChangeText={onFilterData}
+          value={searchTerm}
+          placeholder={placeholderString}
+        />
+        {isScanning ? (
+          <ActivityIndicator
+            style={localStyles.indicatorContainer}
+            color={SUSSOL_ORANGE}
+            size="small"
+          />
+        ) : (
+          <PageButton text={buttonText} onPress={onPress} />
```

Comment:
Just a bit of a lazy way to indicate loading, replacing the button with an activity indicator: https://reactnative.dev/docs/activityindicator. Would be cool to also make a `ButtonWithSpinnerThing` which did this out the box - combined with the `Toast` component mentioned above

Reply:
Was thinking the same thing 😃